src/userdata.rs

```diff
@@ -335,32 +340,38 @@ impl<'lua> AnyUserData<'lua> {
 
     /// Sets an associated value to this `AnyUserData`.
     ///
-    /// The value may be any Lua value whatsoever, and can be retrieved with [`get_user_value`].
+    /// The value may be any Lua value whatsoever, and can be retrieved with [`get_i_user_value`].
     ///
-    /// [`get_user_value`]: #method.get_user_value
-    pub fn set_user_value<V: ToLua<'lua>>(&self, v: V) -> Result<()> {
+    /// [`get_i_user_value`]: #method.get_i_user_value
+    pub fn set_i_user_value<V: ToLua<'lua>>(&self, v: V, n: c_int) -> Result<()> {
```

Comment:
Looking at [this comment](https://github.com/amethyst/rlua/issues/172#issuecomment-616975763) from @kyren, I think the hope is to support multiple versions of Lua simultaneously - I guess it would need to be via Cargo features.
If that is desirable, then I wonder whether it's worth keeping `get_user_value()` and `set_user_value()`.  On Lua 5.4 they would just call e.g. `set_i_user_value(v, 1)` and on earlier versions it would call directly.  the `_i_user_value()` methods would be `#cfg` off on < 5.4 (so that you find out at compile time rather than runtime if you're using a feature which doesn't exist in this version of Lua).

Reply:
Ooh, sure. I'll look into that and see if I can get it working over the next few days. It might be a bit complicated by the API changes that have gone on, though, such as the changes in certain function return types.